pgjdbc/src/main/java/org/postgresql/PGArraySupport.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2004, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql;
+
+import java.sql.Array;
+import java.sql.SQLException;
+
+/**
+ * Defines public PostgreSQL extensions to create {@link Array} objects.
+ */
+public interface PGArraySupport {
+
+  /**
+   * Creates an {@link Array} wrapping <i>elements</i>. This is similar to
+   * java.sql.Connection#createArrayOf(String, Object[]), but also provides
+   * support for primitive arrays.
+   *
+   * @param typeName
+   *          The SQL name of the type to map the <i>elements</i> to.
+   * @param elements
+   *          The array of objects to map.
+   * @return An {@link Array} wrapping <i>elements</i>.
+   * @throws SQLException If for some reason the array cannot be created.
+   * @see java.sql.Connection#createArrayOf(String, Object[])
+   */
+  Array createArrayOf(String typeName, Object elements) throws SQLException;
```

Comment:
This was the nature of my original question. This api is an exact mirror of `java.sql.Connection` with the only change being the support of primitive arrays.
To me, it makes a lot of sense to have that parallel. For folks who have made the existing api work, will continue to be able to make this work.

If there is desire for additional api taking additional attributes, I am willing to add. I need to know what exact api you desire and how to get to the oid for the arguments provided.